test/common/router/config_impl_test.cc

```diff
@@ -2029,5 +2034,229 @@ TEST(RoutePropertyTest, excludeVHRateLimits) {
   EXPECT_TRUE(config_ptr->route(headers, 0)->routeEntry()->includeVirtualHostRateLimits());
 }
 
+TEST(CustomRequestHeadersTest, AddNewHeader) {
```

Comment:
Another quick comment. Please move the majority of the functional tests into a new test file to match the formatter impl. You can have a quick test here if you want just to verify that things are called correctly.